### ISSUE TITLE
S927 FP

### ISSUE THREAD

**Issue by costin-zaharia-sonarsource** [SonarSource/sonar-dotnet#3047]
### Description

False positive for **S927**. When implementing generic interfaces it might be more useful to name the argument depending on the type name than using the generic name from the interface.

### Repro steps

```
private class ConsoleProgressReporter : System.IProgress<ProjectLoadProgress>
{
    public void Report(ProjectLoadProgress loadProgress)
    {

    }
}
```

### Expected behavior

Do not raise the rule.

### Actual behavior

In the previous example I think that **loadProgress** provides more information than *value* used in the generic interface (`void Report(T value);`).

Warning	S927	Rename parameter 'loadProgress' to 'value' to match the interface declaration. (The name of a parameter in an externally visible method override does not match the name of the parameter in the base declaration of the method, or the name of the parameter in the interface declaration of the method or the name of any other partial definition.)

### Related information

* SonarC# Version: 8.2.0.14119
* Visual Studio Version 16.4.3
